specification/securityinsights/resource-manager/Microsoft.SecurityInsights/stable/2024-09-01/DataConnectors.json

```diff
@@ -0,0 +1,1696 @@
+            "type": "string"
+          },
+          "description": "Credentials for service bus namespace, keyvault uri for access key"
+        },
+        "storageAccountCredentialsConfig": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "description": "Credentials for storage account, keyvault uri for access key"
+        }
+      },
+      "x-ms-discriminator-value": "ServiceBus"
+    },
+    "GitHubAuthModel": {
+      "description": "Model for API authentication for GitHub. For this authentication first we need to approve the Router app (Microsoft Security DevOps) to access the GitHub account, Then we only need the InstallationId to get the access token from https://api.github.com/app/installations/{installId}/access_tokens.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/CcpAuthConfig"
+        }
+      ],
+      "type": "object",
+      "properties": {
+        "installationId": {
+          "type": "string",
+          "description": "The GitHubApp auth installation id."
+        }
+      },
+      "x-ms-discriminator-value": "GitHub"
+    },
+    "NoneAuthModel": {
+      "description": "Model for API authentication with no authentication method - public API.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/CcpAuthConfig"
+        }
+      ],
+      "type": "object",
+      "x-ms-discriminator-value": "None"
+    },
+    "JwtAuthModel": {
+      "description": "Model for API authentication with JWT. Simple exchange between user name + password to access token.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/CcpAuthConfig"
+        }
+      ],
+      "properties": {
+        "tokenEndpoint": {
+          "type": "string",
+          "description": "Token endpoint to request JWT"
+        },
+        "userName": {
+          "description": "The user name. If user name and password sent in header request we only need to populate the `value` property with the user name (Same as basic auth). If user name and password sent in body request we need to specify the `Key` and `Value`.",
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          }
+        },
+        "password": {
+          "description": "The password",
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          }
+        },
+        "queryParameters": {
+          "description": "The custom query parameter we want to add once we send request to token endpoint.",
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          }
+        },
+        "headers": {
+          "description": "The custom headers we want to add once we send request to token endpoint.",
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          }
+        },
+        "isCredentialsInHeaders": {
```

Comment:
>isCredentialsInHeaders [](http://example.com/codeflow?start=9&length=22)

same recommendation about using booleans. you can convert it in an enum that could values for where the credential are so that in feature version you can add other values without the need to create other flags

Reply:
same. we reviewed these API in Preview Version. The API in GA for a few months and we dont want to Change the API now